pkg/ensurance/statestore/statstore.go

```diff
@@ -0,0 +1,204 @@
+package statestore
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	ensuranceapi "github.com/gocrane/api/ensurance/v1alpha1"
+	"github.com/gocrane/crane/pkg/ensurance/statestore/nodelocal"
+	"github.com/gocrane/crane/pkg/ensurance/statestore/types"
+	"github.com/gocrane/crane/pkg/utils"
+	"github.com/gocrane/crane/pkg/utils/log"
+)
+
+type stateStoreManager struct {
+	nepInformer cache.SharedIndexInformer
+
+	eventChannel chan types.UpdateEvent
+	index        uint64
+	configCache  sync.Map
+
+	collectors  []collector
+	StatusCache sync.Map
+}
+
+func NewStateStoreManager(nepInformer cache.SharedIndexInformer) StateStore {
+	var eventChan = make(chan types.UpdateEvent)
+	return &stateStoreManager{nepInformer: nepInformer, eventChannel: eventChan}
+}
+
+func (s *stateStoreManager) Name() string {
+	return "StateStoreManager"
+}
+
+func (s *stateStoreManager) Run(stop <-chan struct{}) {
+
+	// check need to update config
+	go func() {
+		updateTicker := time.NewTicker(10 * time.Second)
+		defer updateTicker.Stop()
+		for {
+			select {
+			case <-updateTicker.C:
+				log.Logger().V(4).Info("StateStore config check run periodically")
+				if s.checkConfig() {
+					s.index++
+					log.Logger().V(4).Info("StateStore update event", "index", s.index)
+					s.eventChannel <- types.UpdateEvent{Index: s.index}
+				} else {
+					log.Logger().V(4).Info("StateStore config false, not to update")
+				}
+				continue
+			case <-stop:
+				log.Logger().Info("StateStore config check exit")
+				return
+			}
+		}
+	}()
+
+	// do collect periodically
+	go func() {
+		updateTicker := time.NewTicker(10 * time.Second)
+		defer updateTicker.Stop()
+		for {
+			select {
+			case <-updateTicker.C:
+				log.Logger().V(2).Info("StateStore run periodically")
+				for _, c := range s.collectors {
+					if data, err := c.Collect(); err == nil {
+						for key, v := range data {
+							s.StatusCache.Store(key, v)
+						}
+					} else {
+						log.Logger().Error(err, "StateStore collect failed", c.GetType())
+					}
+				}
+				continue
+			case v := <-s.eventChannel:
+				log.Logger().V(3).Info("StateStore update config index", "Index", v.Index)
+				s.updateConfig()
+			case <-stop:
+				log.Logger().V(2).Info("StateStore exit")
+				return
+			}
+		}
+	}()
+
+	return
+}
+
+func (s *stateStoreManager) List() map[string][]utils.TimeSeries {
+	var maps = make(map[string][]utils.TimeSeries)
+
+	s.StatusCache.Range(func(key, value interface{}) bool {
+		var name = key.(string)
+		var series = value.([]utils.TimeSeries)
+		maps[name] = series
+		return true
+	})
+
+	return maps
+}
+
+func (s *stateStoreManager) AddMetric(key string, t types.CollectType, metricName string, Selector *metav1.LabelSelector) error {
```

Comment:
is AddSimple more accurate for this behavior? you can check predictor code and we can keep same naming convention.